x/auction/types/message_start_auction.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgStartAuction = "start_auction"
+
+var _ sdk.Msg = &MsgStartAuction{}
+
+func NewMsgStartAuction(creator string, zone string, poolID uint64) *MsgStartAuction {
+	return &MsgStartAuction{
+		Creator: creator,
+		Zone:    zone,
+		PoolID:  poolID,
+	}
+}
+
+func (msg *MsgStartAuction) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgStartAuction) Type() string {
+	return TypeMsgStartAuction
+}
+
+func (msg *MsgStartAuction) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgStartAuction) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgStartAuction) ValidateBasic() error {
```

Comment:
We should also check if Zone is valid.
Please check ValidateBasic functions for all messages and see if there are any other missing validation checks.